haystack/reader/farm.py

```diff
@@ -493,33 +493,34 @@ def eval(
                 "context": doc.text
             }
             # get all questions / answers
-            aggregated_per_question: Dict[str, Any] = defaultdict(list)
+            aggregated_per_question: Dict[tuple, Any] = defaultdict(list)
+            key_tuple = (label.id, label.question)
             if doc_id in aggregated_per_doc:
                 for label in aggregated_per_doc[doc_id]:
                     # add to existing answers
-                    if label.question in aggregated_per_question.keys():
+                    if key_tuple in aggregated_per_question.keys():
                         if label.offset_start_in_doc == 0 and label.answer == "":
                             continue
                         else:
                             # Hack to fix problem where duplicate questions are merged by doc_store processing creating a QA example with 8 annotations > 6 annotation max
-                            if len(aggregated_per_question[label.question]["answers"]) >= 6:
+                            if len(aggregated_per_question[key_tuple]["answers"]) >= 6:
```

Comment:
Just seeing this old hack here again: Do I remember correctly that we had a limitation from FARM that we could handle a maximum of 6 annotations per question? Shall we revisit this and either log a warning or allow for more labels? It might become relevant when we work on "answer expansion" where we thought of creating automatically more "answer candidates" from a human seed answer during the annotation process

Reply:
This now logs a warning. I've opened an issue (#1036) to tackle this at a later point.